src/memcache/handler.cpp

```diff
@@ -145,12 +149,27 @@ void handler::on_master_end() {
 }
 
 bool handler::on_slave_start() {
-    int fd = cybozu::tcp_connect(g_config.vip().str().c_str(),
-                                 g_config.repl_port());
+    using logger = cybozu::logger;
+
+    auto master_host = g_config.master_host();
+    int fd;
+    try {
+        fd = cybozu::tcp_connect(master_host.c_str(), g_config.repl_port());
+    } catch( std::runtime_error& err ) {
+        logger::error() << "Failed to connect to the master (" << master_host << "): " << err.what();
+        m_reactor.run_once();
```

Comment:
What does this intend to do? A comment would be helpful.

Reply:
It was called in the existing error handling, so I called it in the code I added.

```cpp
    if( fd == -1 ) {
        m_reactor.run_once();
        return false;
    }
```

I'm not aware of the original intent of the code, so I'm just guessing: since signal handling, etc., depends on the reactor, we should sometimes run the reactor during the re-connecting to the master.

Reply:
I see. Thanks.